railgun/internal/ctrlutils/utils.go

```diff
@@ -59,3 +67,79 @@ func IsAPIAvailable(mgr ctrl.Manager, obj client.Object) (bool, error) {
 
 	return true, nil
 }
+
+// HasAnnotation is a helper function to determine whether an object has a given annotation, and whether it's
+// to the value provided.
+func HasAnnotation(obj client.Object, key, expectedValue string) bool {
```

Comment:
I think that most of these functions added in this file (except for the predicate funcs generator) belong to `pkg/annotations`.

Reply:
I tend to disagree because these functions are commonly operating on `controller-runtime`'s `client.Object`, I don't think introducing that in `pkg/annotations` which previously had no awareness of CR, and which is technically a part of our public API (this package is intentionally under `internal/` to keep us free to move things around later when the dust settles) is something I would like to do within the scope of this PR, I'm assuming this is not a blocker, but if you feel there's more that needs to be said about this please let me know. :bow: